MB - Section Edits (#150)
hassanobeid1994/tr_b_causal_2020: article/sections/_2_graph_importance.tex

\section{Perils of disregard}
\label{sec:graph-importance}

As stated in \ref{sec:intro}, the development of transportation demand models aims at evaluating the impact of policies on a certain
transportation demand related outcome. As an example, consider the proposals from the following fictitious scenarios: 

> - Based on input from the public, the Department of Transportation (DOT) of  is considering implementing a new parking policy. 

Reviewer: ```suggestion
- Based on input from the general public, the Department of Transportation (DOT) is considering implementing a new parking policy. 
```

Reviewer 2: Just a second opinion, I'd stick with the word "general" not being included. After all, who or what is the "specific" public that we are contrasting against when we specify general? I think "public" already connotes the "masses" or the "general population"

Author: +1 to @timothyb0912

This proposed policy would put in place pricing changes and parking restrictions that would discourage individuals from parking in the central business district. 

> The DOT (and its constituents) believes that this policy would encourage people to use more active transportation modes (e.g: walking, biking, etc..).

Reviewer: ```suggestion
The DOT (and its constituents) believes that this policy would encourage people to use more active transportation modes (e.g: walking, biking, etc.).
```

> The DOT (and its constituents) believes that this policy would encourage people to use more active transportation modes (e.g: walking, biking, etc..).

Author: @hassan-obeid does this address your [comment on line 15](https://github.com/hassan-obeid/tr_b_causal_2020/pull/150/files/c65814292b16eae3444b434c69fad2ecca151074?file-filters%5B%5D=.bib&file-filters%5B%5D=.pdf&file-filters%5B%5D=.png&file-filters%5B%5D=.tex&file-filters%5B%5D=No+extension#r532124593)

```suggestion
The DOT (and its constituents) believes that this policy would encourage people to use more active transportation modes (e.g: walking, biking, etc.) and cause them to drive less.
```

- A certain DOT is considering constructing a streetcar (trolley, tram) line in a low/mid-income area in its jurisdiction.
Citing examples from other cities and countries, the DOT claims that the new streetcar line will create more transit oriented
developments and increase economic activity in the areas surrounding the proposed project. 

%% TODO: Present an example or two of real life project proposals?

> Thinking more closely about these proposals, it is clear that they assume a causal relationship between the proposed project/policy and the desired goals. 

Reviewer: ```suggestion
When developing those projects, policy makers implicitly assume a causal relationship between their proposed interventions and the outcomes of interest. An example of such assumed relationships is the following: parking restrictions will cause people to drive less. 
```

Author: I made a change to line 8  that would hopefully address your comment. @hassan-obeid

The DOT in question analyzed data, concluding that such a policy or project would cause the desired output and achieve the desired goal.
In the presented scenarios, the DOT claims that implementing the new parking policy will cause an increase the share of 
active transportation modes in the central business district and that constructing the proposed streetcar line will 

> cause more transit oriented developments and economic activity.

Reviewer: I suggest removing these as they are redundant, considering my suggested changes above.

> The DOT in question analyzed data, concluding that such a policy or project would cause the desired output and achieve the desired goal.
> In the presented scenarios, the DOT claims that implementing the new parking policy will cause an increase the share of 
> active transportation modes in the central business district and that constructing the proposed streetcar line will 
> cause more transit oriented developments and economic activity.

Reviewer: ```suggestion

```

Policymakers base their analyses and conclusions on hypotheses or beliefs of how the world operates.

> In other words, the data analysis is based on a certain belief about the data generating process.

Reviewer: ```suggestion
In other words, the data analysis is based on certain beliefs about the data generating process.
```

> However, policymakers often do not present their beliefs about such data generating process.

Reviewer: ```suggestion
However, policymakers often do not present those beliefs in a clear, concise, and transparent manner.
```

Reviewer 2: What is the difference between clear and transparent? I think they're repetitive in this instance, and I'd instead suggest just "clear and concise manner"

Reviewer: "Transparent" to me has an "honesty" aspect to it. Like the modeler is not trying to hide something. "Clear" is more about not being messy.

Author: +1 to Hassan's suggestion here.

As a result, these proposals maintain an obscure representation of how the policy or project will achieve their desired goals. 

DAGs allow one to clearly encode their assumptions about the data generating process and the problem at hand.
Researchers and practitioners have made use of DAGs in fields ranging from medicine and epidemiology (\citet{shrier_platt_2008}, \citet{sung_2011}) to economics (\citet{white2011causal}) and have found them to be practical.

%% TODO: maybe insert some references of where DAGs were used?.

> Likewise, as in the fields presented above, DAGs could prove useful in addressing transportation policy questions, similar to the ones presented above.

Reviewer: Suggest deleting. The mention to "above" is confusing, since you're referring to both the fields above and the policy questions above, and it's kind of redundant.

> Likewise, as in the fields presented above, DAGs could prove useful in addressing transportation policy questions, similar to the ones presented above.

Reviewer: ```suggestion

```

\citet{brathwaite_2018_causal} have proposed a framework illustrating how practitioners and researchers can use DAGs
 to answer such transportation modeling questions in a causal context.
However, \citet{brathwaite_2018_causal} have not shown an empirical application of their framework and how it results
 in different conclusions when compared to traditional modeling approaches.

> In this section, we will present an example illustrating the importance of using DAGs in transportation demand modeling efforts. 

Reviewer: ```suggestion
In this section, we will present an example illustrating the importance of using DAGs in transportation demand modeling.
```

> Specifically, we will illustrate how different assumptions about the data generating process result in different conclusions.

Reviewer: Added a further edit to your suggestion @hassan-obeid.
```suggestion
Specifically, we will illustrate how different assumptions about the data generating process result in different conclusions, even while assuming the same outcome model.
```

Author: See #163

To make our point, we build upon \citet{brathwaite_2018_causal}.
Here, we present an empirical exercise using a simplified transportation modeling problem.
Before going any further, we note that this example is
illustrative, and it does not reflect all complexities in a typical transportation choice modeling problem.
Indeed, our goal in this section is not to recover the causal effect of the proposed intervention.
Instead, we aim to only show how different DAGs would result in different conclusions about the effect of the proposed intervention.

Let us assume that a company wants to reduce its workforce carbon footprint by moving its employees closer to their campus. 
We would like to forecast how such an intervention would change the share of employees driving to work.
We model this travel mode choice problem based on a dataset from \citet{brathwaite_asymmetric}.
This dataset is based on the 2012 California Household Travel Survey, and it
contains approximately 4000 home-based school or work tours made by approximately 3850 individuals in the California Bay area.
Readers interested in a more detailed description of the dataset can refer to \citet{brathwaite_asymmetric}.

> For purposes of this exercise, we consider the Multinomial Logit model defined in \citet{brathwaite_asymmetric} as the true outcome generating process.

Reviewer: ```suggestion
For purposes of this exercise, we consider the Multinomial Logit model defined in \citet{brathwaite_asymmetric} as the true outcome model.
```

\[ U_{\textrm{Bike}} = ASC_{\textrm{Bike}} + \beta_{\textrm{travel\_distance\_bike}} \times \textrm{Travel\_Distance} \] \\
\[ U_{\textrm{Walk}} = ASC_{\textrm{Walk}} + \beta_{\textrm{travel\_distance\_walk}} \times \textrm{Travel\_Distance} \] \\
\end{aligned}
\end{equation}

For our purposes, we assume that there are no latent variables.
The variables were chosen based on the specification of the MNL model outlined by the equations above.
We can describe the variables as follows:
- Total Travel Distance: is the total travel distance for individual i and mode j, for all available modes for individual i during trip t of tour l.
- Total Travel Cost: is the travel cost in dollars for individual i and mode j, for all available modes for individual i during trip t of tour l.
- Total travel time: is the travel time in minutes for individual i and mode j, for all available modes for individual i during trip t tour l.
- Number of Autos: is the number of automobiles owned by individual i's household.
- Number of Licensed Drivers: is the number of licensed drives in individual i's household.
- Number of Kids: is the number of kids in individual i's household.

> - Cross-bay trip: is a binary variable indicating whether the trip t in tour l for individual i is a cross-bay trip.

Reviewer: ```suggestion
- Total Travel Distance: the total travel distance for individual i and mode j, for all available modes for individual i during trip t of tour l.
- Total Travel Cost: the travel cost in dollars for individual i and mode j, for all available modes for individual i during trip t of tour l.
- Total travel time: the travel time in minutes for individual i and mode j, for all available modes for individual i during trip t tour l.
- Number of Autos: the number of automobiles owned by individual i's household.
- Number of Licensed Drivers: the number of licensed drives in individual i's household.
- Number of Kids: the number of kids in individual i's household.
- Cross-bay trip: a binary variable indicating whether the trip t in tour l for individual i is a cross-bay trip.
```

I suggest either removing the ":", or removing "is"

Author: updated the suggestion to keep ":" and adding to the doc.

Figure \ref{fig:IND_GRAPH} illustrates the DAG where all explanatory variables are independent of each other for each utility equation.
Similarly, Figure \ref{fig:DA_causal_2} through \ref{fig:BIKE_causal_2} illustrate the causal graphs with ``interacting'' explanatory variables.

> Each of these graphs is based on the utility function of each mode in the multinomial logit model specified in \citet{brathwaite_asymmetric}.

Reviewer: I think we need significantly more details here as to why we're assuming each graph. That is, when you reference the graph where all variables are "independent", add some write-up explaining that this assumption is equivalent to the scenario where an analyst only updates the variable directly affected by their policy, without accounting for the relationships between this variable and other explanatory variables. 

The other "interacting" (I would not use this word as it means different things in statistics) represent an assumed "true" data generating process, where variables interact with each other. Emphasis on "assumed true"

Reviewer 2: +1 to Hassan.

I'd say ``dependent'' explanatory variables as opposed to interacting, since the explanatory variables are literally dependent on one another, and it directly contrasts the case of the explanatory variables being independent of one another.

Author: Does this address your comment @hassan-obeid @timothyb0912 

```suggestion
Figure \ref{fig:IND_GRAPH} illustrates the DAG where all explanatory variables are independent of each other for each utility equation. This DAG represents a scenario where an analyst only modifies variables in their dataset that are directly impacted by the proposed intervention, without considering the different relationships between these variables and other explanatory variables in the dataset.
Similarly, Figure \ref{fig:DA_causal_2} through \ref{fig:BIKE_causal_2} illustrate the causal graphs with ``dependent'' explanatory variables. For our purposes, we assume that these DAGs represent the ``true" data generating process, where several variables affect each each other.
Each of these graphs is based on the utility function of each mode in the multinomial logit model specified in \citet{brathwaite_asymmetric}.
```

Reviewer 2: "each each"

Author: See #163

   \includegraphics[width=0.5\textwidth]{WTD_interacting_graph.png}
   \caption{Causal Graph for the Walk-Transit-Drive Utility Function}
   \label{fig:WTD_causal_2}
\end{figure}

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{DTW_interacting_graph.png}
   \caption{Causal Graph for the Drive-Transit-Walk Utility Function}
   \label{fig:DTW_causal_2}
\end{figure}

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{WALK_interacting_graph.png}
   \caption{Causal Graph for the Shared Ride 3+ Utility Function}
   \label{fig:WALK_causal_2}
\end{figure}

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{BIKE_interacting_graph.png}
   \caption{Causal Graph for the Bike Utility Function}
   \label{fig:BIKE_causal_2}
\end{figure}

Using the setup described above, we conduct our simulation exercise as follows:
\begin{enumerate}

\item We build a DAG (Figure \ref{fig:IND_GRAPH}) where we assume all explanatory variables are independent of each other. 
\item We simulate data based on this graph and generate outcomes based on a chosen ``true'' multinomial logit model.
\item We predict outcomes based on the mode choice model.

> \item We apply the do-operator \citet{pearl_causality_2000} to our simulated data to reduce the travel distance for all individuals in the dataset. This emulates a company's decision to move its employees closer to campus.

Reviewer: How is this applying the do-operator?

Reviewer 2: Because it's setting the travel distance?

Author: +1 to @timothyb0912. See this proposed change @hassan-obeid 
```suggestion
\item We apply the do-operator \citet{pearl_causality_2000} to our simulated data by reducing the travel distance for all individuals in the dataset. This emulates a company's decision to move its employees closer to campus.
```

Author: See #163

\item We build a different DAG (Figure \ref{fig:DA_causal_2} through Figure \ref{fig:BIKE_causal_2}) for each utility function based on assumptions of how explanatory variables interact and influence the outcome.
\item We use the real dataset to estimate the relationships between different variables outlined in Figure \ref{fig:DA_causal_2} through Figure \ref{fig:BIKE_causal_2}.
\item We simulate data for variables without parent nodes in Figure Figure \ref{fig:DA_causal_2} through Figure \ref{fig:BIKE_causal_2} and we use the estimated relationships from step 6 to simulate the remaining explanatory variables.
\item We then use the choice model to simulate outcome choices based on data simulated for Figure \ref{fig:DA_causal_2} through Figure \ref{fig:BIKE_causal_2}.
\item We apply the do-operator, as above, to reduce the travel distance for all individuals in the dataset and to emulate a company's decision to move its employees closer to campus.
\item We use the estimated relationships for this causal graph to simulate all the explanatory variables in Figure \ref{fig:DA_causal_2} through Figure \ref{fig:BIKE_causal_2}(namely the variables affected by travel distance).

> \item We use the estimated choice model to produce outcomes based on the simulated data from the previous step.

Reviewer: I think those bullets, the way they're written now, are more confusing than helpful. They're more in the weeds of how the simulation was done mechanically, but they don't convey the bigger picture of what we're trying to accomplish. I suggest replacing with a higher level bullet list that conveys the idea we're trying to illustrate, as opposed to the mechanics. Let's discuss tomorrow.

and a shared ride with two or more individuals).
We compute the difference in mode choice probabilities from different models based on the two constructed DAGs.

We then plot histograms of the computed differences between the average probability of an individual
in our sample choosing a car centric mode before and after implementing a policy or intervention
aimed at reducing travel distance.
These differences are plotted under the two different assumptions about the data generating process illustrated in the causal graphs above.
Figure \ref{fig:histogram_probability} highlights the bias between the estimated probability of an average individual choosing a car centric mode.
This difference shows the importance of considering the data generating process when estimating transportation demand models aiming to forecast the impact of proposed policies.

\begin{figure}
   \centering
   \includegraphics[width=0.5\textwidth]{histogram_selection_on_obs.png}
   \caption{Histograms of the probability of choosing Car Centric Modes under Different Data Generating processes.}
   \label{fig:histogram_probability}
\end{figure}

> The data generating process might not be easily distinguishable in the majority of situations, mainly due to the complexity of the real world.
> Therefore, constructing a causal graph that represents the data generating process as much as possible is not an easy task.
> To prepare readers to create causal graphs, the next section will provide a brief overview of them and their history in choice modelling.
> Then, Section \ref{sec:graph-construction} explores this topic and includes some guidance on how to build causal graphs representing the researchers beliefs about the data generating process.
> Section \ref{sec:graph-testing} follows up with guidance on how to test one's causal graphs against one's data.

Reviewer: ```suggestion
Note that in the simulation presented above, we assumed that the true data generating process is known to the analyst. 
Our goal in doing that is to illustrate how the bias from misrepresenting the data generating process can be arbitrarily large, especially when the analyst takes a naive approach of only focusing on the variables directly impacted by the policy at hand. 
The next section will provide a brief overview of causal graphs and their history in choice modelling.
Then, Section \ref{sec:graph-construction} will provide guidance on how to build causal graphs representing the researchers beliefs about the data generating process.
Section \ref{sec:graph-testing} follows up with guidance on how to test one's causal graphs against one's data.
```

Author: I like your proposed change @hassan-obeid, but I still think that we should re-emphasize how our example is not reflective of what's going on in the real world. I know we mention this in line 40 above, but I think it's worth it to repeat in here. I also thing that the transition from talking about the simulation to talking about the next sections talk about is a little rough. Let's talk today.

Author: See #163
